### ISSUE TITLE
Bug: [prefer-optional-chain] suggestion produces syntax error when used with type assertions

### ISSUE THREAD

**Issue by so1ve** [typescript-eslint/typescript-eslint#11818]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.9.2&fileType=.ts&code=MYewdgzgLgBAZiEMC8MDeBYAUDGAjAQwCcAudAX20q2wSQDJ6YAKOgOkKJkZfc5gIQBYAJ4BKDgQBeY7EA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const foo = {
  bar: {}
}

foo && (foo.bar && (foo.bar as any).baz)
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-optional-chain": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

produces `foo?.bar && (foo.bar as any).baz`

### Actual Result

produces `foo?.bar && (foo.bar as any).baz)`

**Comment by kirkwaiblinger** [comment]
Looks like another simple repro is:

([playground](https://typescript-eslint.io/play/#ts=5.9.3&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPFCgJ4DcAsAFCiSwLLpYBGUAXoceddbkvADIB8ABR8AdKxiDhrNgEogA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uigB3UdCaRwYAL4gNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))

```ts
declare const foo: any;
declare const baz: any;

foo && (foo.bar && baz)
```

Note that there is an unnecessary pair of parentheses in both repros, so this probably doesn't come up often in the wild, but still seems reasonable to fix.
